core/errors/infer.h

```diff
@@ -36,5 +36,6 @@ constexpr ErrorClass UntypedConstantSuggestion{7027, StrictLevel::Strict};
 // constexpr ErrorClass LazyResolve{7029, StrictLevel::True};
 constexpr ErrorClass MetaTypeDispatchCall{7030, StrictLevel::True};
 // N.B infer does not run for untyped call at all. StrictLevel::False here would be meaningless
+constexpr ErrorClass GenericArgumentKeywordArgs{7031, StrictLevel::True};
```

Comment:
This new error is for cases like `T::Array[foo: Integer]`.

Reply:
7031 is in use (PrivateMethod)

You also want to put this before the comment, so that people see it whenever they add a new error to the end.